dockstore-webservice/src/main/java/io/dockstore/webservice/resources/EventResource.java

```diff
@@ -82,22 +83,49 @@ public EventResource(EventDAO eventDAO, UserDAO userDAO) {
+    public List<Event> getUserEvents(@ApiParam(value = "User ID", required = true) @PathParam("userId") Long userId, @QueryParam("event_search_type") EventSearchType eventSearchType, @Min(1) @Max(MAX_LIMIT) @DefaultValue(PAGINATION_DEFAULT_STRING) @ApiParam(defaultValue = PAGINATION_DEFAULT_STRING, allowableValues = PAGINATION_RANGE) @Parameter(schema = @Schema(maximum = "100", minimum = "1")) @QueryParam("limit") int limit, @QueryParam("offset") @DefaultValue("0") Integer offset) {
+        User user = this.userDAO.findById(userId);
+        return getEventsForUser(user, eventSearchType, limit, offset);
+    }
+
+    /**
+     * Returns events for the provided user
+     * @param user The user we are getting events for
+     * @param eventSearchType The types of events
+     * @param limit Event list limit
+     * @param offset Event list offest
+     * @return A list of events
+     */
+    private List<Event> getEventsForUser(User user, EventSearchType eventSearchType, int limit, Integer offset) {
+        if (eventSearchType == null) {
+            return Collections.emptyList();
+        }
+
         switch (eventSearchType) {
         case STARRED_ENTRIES:
-            Set<Long> entryIDs = userWithSession.getStarredEntries().stream().map(Entry::getId).collect(Collectors.toSet());
+            Set<Long> entryIDs = user.getStarredEntries().stream().map(Entry::getId).collect(Collectors.toSet());
             List<Event> eventsByEntryIDs = this.eventDAO.findEventsByEntryIDs(entryIDs, offset, limit);
             eagerLoadEventEntries(eventsByEntryIDs);
             return eventsByEntryIDs;
         case STARRED_ORGANIZATION:
-            Set<Long> organizationIDs = userWithSession.getStarredOrganizations().stream().map(Organization::getId).collect(Collectors.toSet());
+            Set<Long> organizationIDs = user.getStarredOrganizations().stream().map(Organization::getId).collect(Collectors.toSet());
             List<Event> allByOrganizationIds = this.eventDAO.findAllByOrganizationIds(organizationIDs, offset, limit);
             eagerLoadEventEntries(allByOrganizationIds);
             return allByOrganizationIds;
         case ALL_STARRED:
-            Set<Long> organizationIDs2 = userWithSession.getStarredOrganizations().stream().map(Organization::getId).collect(Collectors.toSet());
-            Set<Long> entryIDs2 = userWithSession.getStarredEntries().stream().map(Entry::getId).collect(Collectors.toSet());
+            Set<Long> organizationIDs2 = user.getStarredOrganizations().stream().map(Organization::getId).collect(Collectors.toSet());
+            Set<Long> entryIDs2 = user.getStarredEntries().stream().map(Entry::getId).collect(Collectors.toSet());
             List<Event> allByOrganizationIdsOrEntryIds = this.eventDAO
-                    .findAllByOrganizationIdsOrEntryIds(organizationIDs2, entryIDs2, offset, limit);
+                .findAllByOrganizationIdsOrEntryIds(organizationIDs2, entryIDs2, offset, limit);
```

Comment:
Do we need to combine the organization ids and entry ids somewhere before this?

Reply:
Oh shoot, I read it wrong, disregard.

Reply:
The function takes both as separate parameters (`organizationIDs2` and `entryIDs2`), so this appears correct, unless I'm misunderstanding the question?

Reply:
You are correct.